zocalo/service/dispatcher.py

```diff
@@ -0,0 +1,250 @@
+        if not clean_guid or len(clean_guid) < 3:
+            self.log.warning(
+                "Message with non-conforming guid %s not written to logbook", guid
+            )
+            return
+        try:
+            os.makedirs(os.path.join(basepath, clean_guid[:2]))
+        except OSError:
+            pass  # Ignore if exists
+        try:
+            log_entry = os.path.join(basepath, clean_guid[:2], clean_guid[2:])
+            with open(log_entry, "w") as fh:
+                fh.write("Incoming message header:\n")
+                json.dump(
+                    header,
+                    fh,
+                    sort_keys=True,
+                    skipkeys=True,
+                    default=str,
+                    indent=2,
+                    separators=(",", ": "),
+                )
+                fh.write("\n\nIncoming message body:\n")
+                json.dump(
+                    original_message,
+                    fh,
+                    sort_keys=True,
+                    skipkeys=True,
+                    default=str,
+                    indent=2,
+                    separators=(",", ": "),
+                )
+                fh.write("\n\nParsed message body:\n")
+                json.dump(
+                    message,
+                    fh,
+                    sort_keys=True,
+                    skipkeys=True,
+                    default=str,
+                    indent=2,
+                    separators=(",", ": "),
+                )
+                fh.write("\n\nRecipe object:\n")
+                json.dump(
+                    recipewrap.recipe.recipe,
+                    fh,
+                    sort_keys=True,
+                    skipkeys=True,
+                    default=str,
+                    indent=2,
+                    separators=(",", ": "),
+                )
+                fh.write("\n")
+            self.log.debug("Message saved in logbook at %s", log_entry)
+        except Exception:
+            self.log.warning("Could not write message to logbook", exc_info=True)
+
+    def process(self, header, message):
+        """Process an incoming processing request."""
+        # Time execution
+        start_time = timeit.default_timer()
+
+        # Load processing parameters
+        parameters = message.get("parameters", {})
+        if not isinstance(parameters, dict):
+            # malformed message
+            self.log.warning(
+                "Dispatcher rejected malformed message: parameters not given as dictionary"
+            )
+            self._transport.nack(header)
+            return
+
+        # Unless 'guid' is already defined then generate a unique recipe IDs for
+        # this request, which is attached to all downstream log records and can
+        # be used to determine unique file paths.
+        recipe_id = parameters.get("guid") or str(uuid.uuid4())
+        parameters["guid"] = recipe_id
```

Comment:
suggest a search/replace `guid` -> `uuid`. Best opportunity we have to fix the naming.

Reply:
Done, I think? Will this cause compatibility errors?

Reply:
only in dlstbx, and these need to be addressed when we change DLSDispatcher to inherit from Dispatcher